### PR TITLE
refactor: improve type safety of errors

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Removes needs to cast `Error`s

## How this PR fixes it

Caught errors are typed as `unknown` because they can have any value. As such, we were blindly casting them to `Error`.

A new `asError` function is now used to coerce the `unknown` errors into an `Error` instance accordingly.

## How to test it

Observe all unit tests passing.

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Review by katspaugh — CHANGES_REQUESTED** [review]
Although neat, I'm apprehensive of adding an import every time I need to handle an error just to cast it. We already import `logError` in most of those cases, so maybe we could modify `logError` so that it accepts the entire error object (whatever that might be) and uses `asError` internally.

**Comment by iamacook** [comment]
> We already import `logError` in most of those cases, so maybe we could modify `logError` so that it accepts the entire error object (whatever that might be) and uses `asError` internally.

I've refactored this accordingly, leaving a few as is when the error is needed beyond `logError`.

File: src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx [hunk 1]
```diff
@@ -92,10 +93,12 @@ const ReviewBatchExecute = ({ data, onSubmit }: { data: BatchExecuteData; onSubm
 
     try {
       await (willRelay ? onRelay() : onExecute())
-    } catch (err) {
-      logError(Errors._804, (err as Error).message)
+    } catch (_err) {
+      const err = asError(_err)
+
+      logError(Errors._804, err.message)
```
Comment (katspaugh) [inline src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx:99]: Should it be `logError(Errors._804, err)`?

**Review by katspaugh — COMMENTED** [review]
There's a few places where you're still passing a message to logError instead of the thrown object.

**Review by katspaugh — APPROVED** [review]
Thanks! 👍

**Comment by iamacook** [comment]
Closing in favour of https://github.com/safe-global/safe-wallet-web/pull/2127.

**Pull request closed without merging**